include/fmt/chrono.h

```diff
@@ -793,11 +793,16 @@ struct chrono_formatter {
 
   explicit chrono_formatter(FormatContext& ctx, OutputIt o,
                             std::chrono::duration<Rep, Period> d)
-      : context(ctx), out(o), val(d.count()), negative(false) {
-    if (d.count() < 0) {
-      val = 0 - val;
+      : context(ctx), out(o), val(static_cast<rep>(std::abs(d.count()))), negative(d.count() < 0) {
+
+   if (d.count() < 0) {
+      val = static_cast<rep>(- d.count());
       negative = true;
-    }
+   }
+   else {
+      val = static_cast<rep>(d.count());
+      negative = false;
+   }
```

Comment:
There is too much redundancy here. Please keep the old code and just add a cast.

Reply:
I apologize for the line 796, it was mismerged by me. I intended to remove the member initialization of val and negative.

But we can't keep the old code, since it assigns a possibly signed value "d.count()" into an unsigned variable "val". You need "val" to acquire the absolute value of d.count, which can be cleanly done either with "std::abs()" like on line 796, or with the test on 798:804.

Which one would you prefer?

Reply:
I think it should be enough to add `static_cast` to the initializer list.